src/mz/src/regions.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::utils::trim_newline;
+/// ----------------------------
+///  Regions commands
+/// ----------------------------
+use crate::{CloudProvider, CloudProviderRegion, FronteggAuthMachine, Region, CLOUD_PROVIDERS_URL};
+
+use std::{collections::HashMap, io::Write};
+
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, CONTENT_TYPE, USER_AGENT};
+use reqwest::{Client, Error, Response};
+
+fn parse_cloud_provider_region(cloud_provider_region: CloudProviderRegion) -> String {
+    match cloud_provider_region {
+        CloudProviderRegion::usEast_1 => "us-east-1".to_string(),
+        CloudProviderRegion::euWest_1 => "eu-west-1".to_string(),
+    }
+}
+
+// TODO: ec.0 is dynamic.
+fn format_region_url(cloud_provider_region: CloudProviderRegion) -> String {
+    format!(
+        "https://ec.0.{}.aws.cloud.materialize.com/api/environment",
+        parse_cloud_provider_region(cloud_provider_region)
+    )
+}
+
+fn build_region_request_headers(authorization: String) -> HeaderMap {
+    let mut headers = HeaderMap::new();
+    headers.insert(USER_AGENT, HeaderValue::from_static("reqwest"));
+    headers.insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
+    headers.insert(
+        AUTHORIZATION,
+        HeaderValue::from_str(authorization.as_str()).unwrap(),
+    );
+
+    headers
+}
+
+pub(crate) async fn enable_region(
+    client: Client,
+    cloud_provider_region: CloudProviderRegion,
+    frontegg_auth_machine: FronteggAuthMachine,
+) -> Result<Region, reqwest::Error> {
+    let authorization: String = format!("Bearer {}", frontegg_auth_machine.access_token);
+    let region_url: String = format_region_url(cloud_provider_region);
+
+    let headers = build_region_request_headers(authorization);
+    let mut body = HashMap::new();
+    // TODO: Version is dynamic
+    body.insert(
+        "coordd_image_ref",
+        &"materialize/environmentd:unstable-45e2acde087c27a661b5e67db587375c0b628fde",
+    );
```

Comment:
I would just *not* set this now, and that'll be good enough. The EC recently learned to prefer that clients say nothing about what version to use, and it'll automatically pick the latest release.